ambry-store/src/main/java/com.github.ambry.store/ScanResults.java

```diff
@@ -0,0 +1,194 @@
+package com.github.ambry.store;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+
+/**
+ * Hold the data structures needed by {@link BlobStoreStats} to serve requests. The class also holds helper methods
+ * used to modify and access the stored data structures.
+ */
+class ScanResults {
+  private final long bucketCount;
+  private final long bucketTimeSpan;
+  private final SortedMap<String, Long> requestedSegmentMap = new TreeMap<>();
+  private final HashMap<String, HashMap<String, Long>> containerBaseMap = new HashMap<>();
```

Comment:
There is no reason to declare this as a `HashMap` is there? If you are not using APIs specific to the implementation, it is always best to use the interface when declaring (`Map` in this case - if this was a `TreeMap` where properties of `NavigableMap` were being used, it would have been `NavigableMap`). This applies for arguments to functions and return types from functions too. From a cursory look, there are a quite few places that need changes. Could you take a look and make changes wherever required?